### PR TITLE
Reduce C# emit test allocations

### PR DESCRIPTION
**Pull request by agocke**
This change reduces the VM used and memory allocated in the emit
tests by moving the WinRT tests to their own DLL and removing
the extra mscorlib reference.

The cause of most of the allocations in the Emit unit tests was
serializing the WinRT assemblies across app domains. There were
three reasons the DLLs needed to be serialized:

1) PeVerify currently requires a separate app domain to be used if
   any of the references of the assembly to be verified have already
   been loaded into the current app domain and have a different MVID.

2) The WinRT tests all use a separate version of mscorlib, so they
   will all reference an assembly with a different MVID than the one
   loaded.

3) Most of the WinRT tests were accidentally pulling two references
   to mscorlib, also always leading to a conflict.

All of these issues should be resolved. The CSharp.Emit unit tests
now allocated 400 MB less and have > 100 MB smaller peak VM.

### CODE REVIEW

**Comment by agocke** [comment]
@jaredpar

File: src/Compilers/CSharp/Test/WinRT/CSharpWinRTTest.csproj [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    <WarningLevel>4</WarningLevel>
+  </PropertyGroup>
+  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Release|AnyCPU' ">
+    <DebugType>pdbonly</DebugType>
+    <Optimize>true</Optimize>
+    <OutputPath>bin\Release\</OutputPath>
+    <DefineConstants>TRACE</DefineConstants>
+    <ErrorReport>prompt</ErrorReport>
+    <WarningLevel>4</WarningLevel>
+  </PropertyGroup>
```
Comment (jaredpar) [inline src/Compilers/CSharp/Test/WinRT/CSharpWinRTTest.csproj]: I think the two above sections could be simplified to 

``` xml
  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Debug|AnyCPU' /">
  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Release|AnyCPU" />
```

This seems to be the approach most of our other projects take.

**Comment by jaredpar** [comment]
Looks great!

Please make sure to update the internal solution file as well for the new project after checking in.

**Comment by agocke** [comment]
@dotnet-bot  please build this

**Comment by agocke** [comment]
Jenkins test this please

**Comment by agocke** [comment]
retest this please

**Pull request merged**